### DISCUSSION TITLE
Why span not label?

### DISCUSSION THREAD

**Discussion by polRk** [radix-ui/primitives#1025] [Show and tell]
https://github.com/radix-ui/primitives/blob/55a422c5830a52d9f050369412a7dc8cf6e44786/packages/react/label/src/Label.tsx#L61

**Comment by jjenzz** [comment]
It's a label that can be used for labelling anything, not just native form controls e.g. custom controls. It was historically needed to provide appropriate labelling to our custom `Checkbox`, `Radio` and `Switch` components but they have since been improved to support native `<label />`. 

I wonder if our `Label` is redundant nowadays 🤔

**Reply by zeyuri** [reply]
@jjenzz Do you think a change in the Label component is needed? And if so, the maintainers are open to a PR? I really want to contribute to this amazing project.

**Reply by jjenzz** [reply]
@zeyuri Great to hear that you'd like to get involved. We're always open to PRs but discussing first is definitely the right approach as I think it might be a little premature to make changes to the `Label` at this stage. 

We have plans in the future to build out a `Form` component which might include a `Form.Label` that renders `label`s. That work might deprecate the existing `Label` or give it a rename but I can't know till that work is begun unfortunately.
